src/ImageSharp/Formats/Jpeg/Components/Block8x8F.cs

```diff
@@ -663,213 +608,56 @@ public override string ToString()
-                d.V1R.X = this.V4L.Y;
-                d.V2R.X = this.V4L.Z;
-                d.V3R.X = this.V4L.W;
-                d.V4R.X = this.V4R.X;
-                d.V5R.X = this.V4R.Y;
-                d.V6R.X = this.V4R.Z;
-                d.V7R.X = this.V4R.W;
-
-                d.V0R.Y = this.V5L.X;
-                d.V1R.Y = this.V5L.Y;
-                d.V2R.Y = this.V5L.Z;
-                d.V3R.Y = this.V5L.W;
-                d.V4R.Y = this.V5R.X;
-                d.V5R.Y = this.V5R.Y;
-                d.V6R.Y = this.V5R.Z;
-                d.V7R.Y = this.V5R.W;
-
-                d.V0R.Z = this.V6L.X;
-                d.V1R.Z = this.V6L.Y;
-                d.V2R.Z = this.V6L.Z;
-                d.V3R.Z = this.V6L.W;
-                d.V4R.Z = this.V6R.X;
-                d.V5R.Z = this.V6R.Y;
-                d.V6R.Z = this.V6R.Z;
-                d.V7R.Z = this.V6R.W;
-
-                d.V0R.W = this.V7L.X;
-                d.V1R.W = this.V7L.Y;
-                d.V2R.W = this.V7L.Z;
-                d.V3R.W = this.V7L.W;
-                d.V4R.W = this.V7R.X;
-                d.V5R.W = this.V7R.Y;
-                d.V6R.W = this.V7R.Z;
-                d.V7R.W = this.V7R.W;
+                this.TransposeScalar();
             }
         }
 
         /// <summary>
-        /// Compares entire 8x8 block to a single scalar value.
+        /// Scalar inplace transpose implementation for <see cref="Transpose"/>
         /// </summary>
-        /// <param name="value">Value to compare to.</param>
-        public bool EqualsToScalar(int value)
+        [MethodImpl(InliningOptions.ShortMethod)]
+        private void TransposeScalar()
         {
-#if SUPPORTS_RUNTIME_INTRINSICS
-            if (Avx2.IsSupported)
-            {
-                const int equalityMask = unchecked((int)0b1111_1111_1111_1111_1111_1111_1111_1111);
-
-                var targetVector = Vector256.Create(value);
-                ref Vector256<float> blockStride = ref this.V0;
-
-                for (int i = 0; i < RowCount; i++)
-                {
-                    Vector256<int> areEqual = Avx2.CompareEqual(Avx.ConvertToVector256Int32WithTruncation(Unsafe.Add(ref this.V0, i)), targetVector);
-                    if (Avx2.MoveMask(areEqual.AsByte()) != equalityMask)
-                    {
-                        return false;
-                    }
-                }
+            float tmp;
+            int horIndex, verIndex;
 
-                return true;
-            }
-#endif
+            // We don't care about the last row as it consists of a single element
+            // Which won't be swapped with anything
+            for (int i = 0; i < 7; i++)
             {
-                ref float scalars = ref Unsafe.As<Block8x8F, float>(ref this);
-
-                for (int i = 0; i < Size; i++)
+                // We don't care about the first element in each row as it's not swapped
+                for (int j = i + 1; j < 8; j++)
```

Comment:
Can't we replace the loop with hard coded unsafe pointer code? Or maybe bring back the old `d.V0L.X = this.V0L.X;` style code?

Reply:
It's actually slower but not that much:
|            Method |                Job |      Mean |     Error |    StdDev |
|------------------ |------------------- |----------:|----------:|----------:|
| OLD TransposeInto | 1. No HwIntrinsics | 14.558 ns | 0.0834 ns | 0.0739 ns |
| NEW TransposeInto | 1. No HwIntrinsics | 19.897 ns | 0.0661 ns | 0.0619 ns |

While old approach is better due to almost direct access and no branching due to loops, it can be optimized by not assigning cells from main diagonal (which what code in this PR does). I think `Unsafe.Add(ref blockRef, index)` is better as it doesn't depend on `Vector4` interface of the `Block8x8F`.

Reply:
Yeah, "swapping" the diagonal values was a mistake, glad you noticed that.

Depending how much we care about fallback code, I think the ~30% difference shown by your benchmarks *might* be significant, so if you see a way to improve this without spending too much time, that would be great.

Note that I implemented this in a non-inplace (`TransposeInto`) form to avoid moving data to/from a `tmp` variable. I wonder how much impact it has.

Reply:
I'd like to avoid a regression if we can since this will be the last version for legacy platforms.

Reply:
It doesn't depend on the `Vector` intrinsics I was just to lazy to unroll this loop :P
I will fix this in the following commit(s).

Reply:
Legend... Thanks.